api/http/handler/auth/authenticate.go

```diff
@@ -58,20 +58,20 @@ func (handler *Handler) authenticate(w http.ResponseWriter, r *http.Request) *ht
 
 	if settings.AuthenticationMethod == portainer.AuthenticationLDAP {
 		if u == nil && settings.LDAPSettings.AutoCreateUsers {
-			return handler.authenticateLDAPAndCreateUser(w, payload.Username, payload.Password, &settings.LDAPSettings)
+			return handler.authenticateLDAPAndCreateUser(w, payload.Username, payload.Password, &settings.LDAPSettings, settings.UserSessionTimeout)
```

Comment:
Instead of passing around the UserSessionTimeout parameter everywhere in there I think it should be injected directly into the JWTService.